docs/concepts/json_schema.md

```diff
@@ -617,6 +617,72 @@ print(json.dumps(Model.model_json_schema(), indent=2))
 """
 ```
 
+#### Merging `json_schema_extra`
+
+Starting in v2.9, Pydantic merges `json_schema_extra` dictionaries from annotated types.
+This pattern offers a more additive approach to merging rather than the previous override behavior.
+This can be quite helpful for cases of reusing json schema extra information across multiple types.
+
+We viewed this change largely as a bug fix, as it unifies the json schema extra merging behavior
+between `BaseModel` and `TypeAdapter` instances - see [this issue](https://github.com/pydantic/pydantic/issues/9210)
+for more details.
```

Comment:
I'd change `unifies the json schema extra merging behavior` to `resolves unintentional differences in the json schema extra merging behavior`.